ray-operator/controllers/ray/common/service_test.go

```diff
@@ -78,4 +78,29 @@ func TestBuildServiceForHeadPod(t *testing.T) {
 	if !reflect.DeepEqual(expectedResult, actualResult) {
 		t.Fatalf("Expected `%v` but got `%v`", expectedResult, actualResult)
 	}
+
+	actualResult = svc.Spec.Selector[KubernetesApplicationNameLabelKey]
+	expectedResult = ApplicationName
+	if !reflect.DeepEqual(expectedResult, actualResult) {
+		t.Fatalf("Expected `%v` but got `%v`", expectedResult, actualResult)
+	}
+}
+
+func TestBuildServiceForHeadPodWithDefaultLabel(t *testing.T) {
+	labels := make(map[string]string)
+	labels[KubernetesApplicationNameLabelKey] = "testname"
+	svc, err := BuildServiceForHeadPod(*instanceWithWrongSvc, labels)
+	assert.Nil(t, err)
+
+	actualResult := svc.Spec.Selector[KubernetesApplicationNameLabelKey]
+	expectedResult := "testname"
+	if !reflect.DeepEqual(expectedResult, actualResult) {
+		t.Fatalf("Expected `%v` but got `%v`", expectedResult, actualResult)
+	}
+
+	actualLength := len(svc.Spec.Selector)
```

Comment:
Could you add a brief comment explaining this branch of test logic?
(for example, mention that 5 is the number of default labels, and that we're checking that we've correctly modified the app name label without adding any extra labels)

Reply:
Updated.